htdocs/core/class/commonobject.class.php

```diff
@@ -5988,6 +5988,12 @@ public function insertExtraFields($trigger = '', $userused = null)
 							dol_syslog('Error bad setup of extrafield', LOG_WARNING);
 						}
 						break;
+					case 'checkbox':
+					case 'chkbxlst':
+						if (is_array($this->array_options[$key])) {
+							$new_array_options[$key] = implode(',', $this->array_options[$key]);
```

Comment:
And what happen if it is not array($this->array_options[$key])). I think you miss the 
else
`$new_array_options[$key] = $this->array_options[$key]`